src/ChannelServer/Network/ChannelPacketSender.cs

```diff
@@ -879,7 +879,7 @@ public static void ZC_OBJECT_PROPERTY(ChannelConnection conn, Character characte
 			packet.PutLong(character.Id);
 			foreach (var property in properties)
 			{
-				packet.PutShort(property);
+				packet.PutInt(property);
 				switch (property)
 				{
 					case ObjectProperty.PC.HP: packet.PutFloat(character.Hp); break;
```

Comment:
As an aside, is there a reason you did not choose to make `ObjectProperty` a dictionary? I found instances where I needed to iterate through a list of these and was forced to write a switch statement like the one above. When we move toward sending a list of all stats for a character (I think 50-60 stats), then these statements will get lengthy.

That's just an observation I wanted to make. If you have a good reason for them being constant variables, then just let me know.

Reply:
You mean like `Dictionary<ObjectProperty, float>`? The problem with that is that not all properties are floats as far as I know, there are at least also ints and strings. As a result you would need do make it `<ObjectProperty, object>`, and that would result in a lot of boxing, unboxing, and type checking.

Tachiorz was working on an alternative way of handling properties back when we started, though that solution has problems in addition to the boxing. https://github.com/aura-project/melia/issues/53

Reply:
I should be more clear. The class named `ObjectProperty.cs` looks likes this.
```
public static class ObjectProperty
	{
		public static class Account
		{
			public const short MaxPVPGrade = 6299;
			public const short Medal = 6300;
                        ...
                }
        }
```

I'm suggesting something such as this.
```
public static class ObjectProperty 
{
    public static readonly Dictionary<string, int> Account = new Dictionary<string, int>
    {
        { "MaxPVPGrade", 6299 },
        ...
    }
}
````

Reply:
Oh, well, that would be highly inefficient. Every access to one property id would entail a look up in a dictionary.  You would also lose intellisense and value safety for the names, unless you add another class just for the names.

In case you also want to know why it's consts and not an enum, that's because we don't actually need enumeration for them, and this way we don't have to cast all the time.

To still iterate over such a class you can actually use reflection though.
```csharp
foreach (var constant in typeof(ObjectProperty.Account).GetFields())
{
	Log.Debug(constant.Name + ": " + constant.GetValue(null));
}
```